updated version of ufs to include WoFS suite and ccpp-v6

mark-a-potts opened the pull request:
This change updates the version of the UFS Weather Model to a cherry-picked version based off the hash from April 4 but also includes the WoFS suite.

Commit:
Fix artifact creation for branches with / in name (#275)

The slash in branch names such as release/.* and feature/.* is
interpreted as a directory separator. This change replaces the "/"
character with a "_".

gspetro-NOAA committed:
Include documentation for building/running SRW App on Mac (#240)

* updated docs

* added git submodule

* fix formatting

* added new submodule commits

* fixed ref links

* finished Intro

* finish Components & Intro edits

* edited Rocoto workflow section of Quickstart

* added minor hpc submodule commits

* Updates to Rocoto Workflow in Quick Start

* add to HPC-stack intro

* submodule updates

* added submodule docs edits

* hpc-stack updates & formatting fixes

* hpc-stack intro edits

* bibtex attempted fix

* add hpc-stack module edits

* update sphinxcontrib version

* add .readthedocs.yaml file

* update .readthedocs.yaml file

* update .readthedocs.yaml file

* update conf.py

* updates .readthedocs.yaml with submodules

* updates .readthedocs.yaml with submodules

* submodule updates

* submodule updates

* minor Intro edits

* minor Intro edits

* minor Intro edits

* submodule updates

* fixed typos in QS

* QS updates

* QS updates

* QS updates

* updates to InputOutput and QS

* fix I/O doc typos

* pull updates to hpc-stack docs

* pull updates to hpc-stack docs

* fix table wrapping

* updates to QS for cloud

* fix QS export statements

* fix QS export statements

* QS edits on bind, config

* add bullet points to notes

* running without rocoto

* add HPC-Stack submodule w/docs

* split QS into container/non-container approaches

* added filepath changes for running in container on Orion, et al.

* edits to overview and container QS

* moved CodeReposAndDirs.rst info to the Introduction & deleted file

* continued edits to SRWAppOverview

* combine overview w/non-container docs

* finish merging non-container guide & SRWOverview, rename/remove files, update FAQ

* minor edits for Intro & QS

* updates to BuildRun doc through 3.8.1

* edits to Build/Run and Components

* remove .gitignore

* fix Ch 3 title, 4 supported platform levels note

* fix typos, add term links

* other minor fixes/suggestions implemented

* updated Intro based on feedback; changed SRW to SRW App throughout

* update comment to Intro citation

* add user-defined vertical levels to future work

* Add instructions for srw_common module load

* fix typo

* update Intro & BuildRunSRW based on Mark's feedback

* minor intro updates

* 1st round of jwolff's edits

* 2nd round of jwolff updates

* update QS intro

* fix minor physics details

* update citation and physics suite name

* add compute node allocation info to QS

* add authoritative hpc-stack docs to Intro

* create MacOS install/build instructions

* add MacOS Build/Run instructions

* fix MacOS Build/Run details

* add MacOS info directly to Build/Run SRW chapter

* minor details

* minor edits

* update Include-HPCInstall with mac installation docs

* add note re: Terminal.app & bash shell in MacOS section

* remove MacInstall file-contents added to BuildRunSRW

* update hpc-stack submodule to include mac installation info

* add MacOS config details

* add MacOS config & run details

* minor MacOS note

* mention need to verify software library version #'s

* update hpc-stack-mod

* align MacDetails section with PR #238 info

* remove gsed & alter related commands

* update hpc-stack submodule

* typos

* switch from env to module load

Co-authored-by: Will Mayfield <59745143+willmayfield@users.noreply.github.com>

* Update BuildRunSRW.rst

* update hpc-stack module docs & MacOS config.sh

* update machine file instructions

* updates to BuildRun chapter

* fix typo

Co-authored-by: gspetro <gillian.s.petro@gmail.com>
Co-authored-by: Will Mayfield <59745143+willmayfield@users.noreply.github.com>

gsketefian committed:
Update documentation for CSV file containing WE2E test info (#278)

* Edits to documentation to match latest in code.

* Edits to documentation to match latest in code.

* Minor changes to documentation.

chan-hoo committed:
Port SRW App to WCOSS2 (#270)

* port to wcoss2

* update scripts

* ensure platform name from variance

* update scripts

* remove wcoss2 from lmod-setup

chan-hoo committed:
Fix --ccpp option in devbuild.sh (#280)

* fix ccpp issue

* update script

danielabdi-noaa committed:
Print usage if machine name is not passed to `lmod-setup.sh/csh`. (#262)

* Print usage() message if machine name is not passed.

* Bug fix with macos modulefiles.

chan-hoo committed:
Back to original hashes (#287)

natalie-perlin committed:
Updated the Introduction, build for MacOS (#281)

* Updated the Introduction, build for MacOS

* Update Introduction.rst

Some comments removed

* Update Introduction.rst

Updated data storage requirements for the SRW

JeffBeck-NOAA commented on Externals.cfg:
```diff
@@ -18,19 +18,19 @@ required = True
 protocol = git
-repo_url = https://github.com/ufs-community/ufs-weather-model
+repo_url = https://github.com/NOAA-EPIC/ufs-weather-model-1
 # Specify either a branch name or a hash but not both.
```
@mark-a-potts, should this point to ufs-community and not NOAA-EPIC?  Or is this just for testing?

mark-a-potts replied:
Just for testing right now.

mark-a-potts replied:
I tried to update the pointers to ccpp/physics and ccpp/framework, but that broke the build and I had to revert to the hashes we had on April 4. I am worried that that means we will not have release/public-v6 of ccpp in our release. Is this a reason to delay?

JeffBeck-NOAA replied:
OK, we'll just have to see if the WoFS SDF XML will work without the pointer updates.  I've emailed @ligiabernardet  and @grantfirl to ask about the status of the v6 CCPP release branch and whether it includes the major module updates that went into the ufs-weather-model repo recently.

mark-a-potts replied:
Thanks, @JeffBeck-NOAA  I emailed Grant as well. He had commits to release-v6 today, so I am thinking that even if they are ready on their side, the changes at the fv3atm level have not been merged yet. @arunchawla-NOAA recommended on slack that we should probably delay the release of the SRW. I am on the fence, but if we do the release, I don't think it can be with ccpp-v6.

JeffBeck-NOAA replied:
@mark-a-potts, just read through the email threads related to the physics/fv3atm release branches.  I would agree with your assessment that it's best to delay by about a week.  I'd hope it won't take too long to get the branch situation straightened out, but there are several submodule branches/pointers that have to be lined up with the correct cherry-picked PRs.  It looks like Grant has taken a big step already toward sorting this out.  However, I'm wondering if the resulting release branch tree will still require the major module updates that currently don't work in the App.  If that does turn out to be the case, I'm happy to quickly track down the required changes to srw_common and get a PR open into the App-level release branch.

mark-a-potts replied:
I agree. Let's postpone the release and focus on getting the physics/branch/submodule situation lined up to everyone's satisfaction. I think we add considerable value to the release by incorporating the latest physics package.

gsketefian committed:
Update documentation for python plotting scripts (#289)

## DESCRIPTION OF CHANGES: 
This PR updates the RST documentation files to accompany changes in PR #[783](https://github.com/ufs-community/regional_workflow/pull/783) in `regional_workflow`.

## DEPENDENCIES:
PR #[783](https://github.com/ufs-community/regional_workflow/pull/783) in `regional_workflow`.

## ISSUE (optional): 
Partially resolves issue #[781](https://github.com/ufs-community/regional_workflow/issues/781) (still need to introduce tests in `regional_workflow` that run the plotting scripts).

## CONTRIBUTORS: 
@mark-a-potts

mark-a-potts committed:
Updating for testing with updated ufs weather model

mark-a-potts committed:
added version to wflow for jet

mark-a-potts committed:
switched to epic repo for testing

mark-a-potts committed:
Updates to modules for cheyenne

mark-a-potts committed:
updated module path

mark-a-potts committed:
changed hash to branch for testing

gsketefian commented on devbuild.sh:
```diff
@@ -136,6 +136,11 @@ while :; do
   shift
 done
 
+# Ensure uppercase / lowercase ============================================
```
FYI the `^^` and `,,` bashisms don't work on a Mac because Apple apparently doesn't/can't update its version of bash on MacOs (and whatever version they're frozen on doesn't support these features).  That's why in `regional_workflow`, we instead use the `echo_uppercase` and `echo_lowercase` functions, e.g. `COMPILER=$(echo_lowercase $COMPILER)`.  To be able to do that, you have to first source `source_util_funcs.sh` in `regional_workflow/ush`.

mark-a-potts committed:
updated to use ufs-community version of WM

mark-a-potts committed:
Pointing to PR-branch of regional_workflow for testing

mark-a-potts committed:
Merge branch 'release/public-v2' into feature/ufs-update

mark-a-potts committed:
Updating module files for cheyenne for testing

mark-a-potts committed:
Added png explicitly under gaea.

natalie-perlin committed:
Update build_cheyenne_intel

Corrected loading system modules (load intel/2022.1, mpt/2.25), updated package versions if newer modules are built.

mark-a-potts committed:
Updated pointer to model for testing

mark-a-potts committed:
fixed repos

mark-a-potts committed:
updated to point at my regional_workflow branch

mark-a-potts committed:
updated regional_workflow to release/public-v2
